src/components/panel/panel.js

```diff
@@ -865,6 +865,15 @@ MdPanelRef.prototype.close = function() {
         return self.detach();
       }, this._$q.reject)
       .then(function() {
+        var onDomRemoved = self._config['onDomRemoved'];
+
+        if (onDomRemoved) {
+          return self._$q.when(onDomRemoved());
+        } else {
+          return true;
```

Comment:
Shouldn't you return a promise in both cases so that chaining won't be broken.

I think this should be: `self._$q.resolve(self)`